explorer/src/components/SearchBar.tsx

```diff
@@ -33,9 +46,26 @@ export function SearchBar() {
   };
 
   const onInputChange = (value: string, { action }: InputActionMeta) => {
-    if (action === "input-change") setSearch(value);
+    if (action === "input-change") {
+      setSearch(value);
+    }
   };
 
+  React.useEffect(() => {
```

Comment:
This works pretty well but could you make the following changes:
- Change this to only load domain search results (domain search results can be appended to the local search results)
- Add a loading state
- Don't set the domain search result if `search` changed during the lookup

Reply:
When `search` changes, this code:
1. sets the search bar to the default loading state
2. rebuilds local options and updates temp search
3. checks if the search input is a domain and, if so, triggers a new domain lookup but otherwise disables loading

The domain lookup will:
1. update the loading message
2. async builds domain search options
3. updates temp search
4. disables loading

Whenever the temp search options change and they match the current search term, the actual search options will be updated and shown to the user. They will only not match the current search term if `search` changed before the async domain lookup finishes.

This works quite well! But I think there is still a race condition where the search term changes while one domain request is still running because the loading state will get set to true and we no longer care if the previous domain request finishes running.

I would suggest creating mutable ref object with `React.useRef` to track whenever `search` gets updated. When a domain request starts you can save the ref's current value and compare it to the value when the request finishes. If they're different, you don't need update temp search options or the loading state because we know another search request has started.